kglib/kgcn/test/base.py

```diff
@@ -0,0 +1,70 @@
+#
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+#
+
+import os
+import shutil
+import subprocess as sp
+import tempfile
+import zipfile
+
+
+class ZipFile(zipfile.ZipFile):
+    """ Custom ZipFile class handling file permissions. """
+    def _extract_member(self, member, targetpath, pwd):
+        if not isinstance(member, zipfile.ZipInfo):
+            member = self.getinfo(member)
+
+        targetpath = super()._extract_member(member, targetpath, pwd)
+
+        attr = member.external_attr >> 16
+        if attr != 0:
+            os.chmod(targetpath, attr)
+        return targetpath
+
+
+class GraknServer(object):
```

Comment:
let's rename it to `GraknCore`? It's more descriptive because it is the product name.

Reply:
you can do this in the next PR

Reply:
This would be inconsistent with the implementation in `client-python`. In general we should have one such implementation, probably in `bazel-distribution` as per @vmax 's suggestion, and depend upon that here.

Reply:
I see. `bazel-distribution` is not the place for this. `build-tools` is also not appropriate in my opinion.

Should we just have one in `client-python` (e.g., `@graknlabs_client_python//test/lib:test-server`) that can be used from here? cc: @vmax